infra/azure/terraform/README.md

```diff
@@ -0,0 +1,47 @@
+# infra/azure
+Terraform scripts and documentation for Cluster API for Azure ([CAPZ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/templates)) infrastructure that the Kubernetes community runs on Azure.
+The capz folder structure is formatted shown below:
+main.tf
+├── resource_group
+├── identities
+├── key_vault
+├── container_registry
+├── storage_account
+├── role_assignments
+├── variables
+
+# Prerequiste 
+The `az-cli-prow` service principal with federated credentials is required to be created by a tenant admin. The [iam-config\gce-param.json](./iam-config/gce-param.json) and [iam-config\eks-param.json](./iam-config/eks-param.json) can be used for creating the fededrated credentials for GCE and EKS.
+
+The service principal needs:
+- Contributor role access to the sub. 
+- Creation of a custom role to give write access
+- acrpush role for the registry 
+- Storage Blob Data Contributor role for Storage account

```

Comment:
can we update this to be clear that these three are managed by terrafrom?

Also add commands to apply the Federated credentials/access to sub mentioned above 

```
az ad sp create-for-rbac --name az-cli-prow
appid=$(az ad sp list --filter "displayName eq 'az-cli-prow'" --query [0].appId --output tsv)
az ad app federated-credential create --id $appid --parameters param.json

objectid=$(az ad sp list --filter "displayName eq 'az-cli-prow'" --query [0].id --output tsv)
az role assignment create --assignee-object-id $(objectid) --assignee-principal-type ServicePrincipal --role Contributor --scope /subscriptions/<subid>
```

Reply:
The contributor role is added by our scripts, provided the person who runs them is a contributor to the subscription.